### PR TITLE
Add task debug info in job detail page [webportal]

### PR DESCRIPTION
**Pull request by debuggy**
image

### CODE REVIEW

File: src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx [hunk 1]
```diff
@@ -321,25 +323,115 @@ export default class TaskRoleContainerList extends React.Component {
...
+            <div className={c(FontClassNames.mediumPlus)}>
+              {item.podNodeName}
+            </div>
+          );
+        },
+      },
+      {
+        key: 'exitCode',
+        name: 'Exit Code',
+        headerClassName: FontClassNames.medium,
```
Comment (yqwang-ms) [inline src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx:346]: By default, also show exitCode, exitType, exitDiagonostic, startTime and completionTime?

File: src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx [hunk 2]
```diff
@@ -490,6 +627,21 @@ export default class TaskRoleContainerList extends React.Component {
         isResizable: true,
         onRender: item => <StatusBadge status={capitalize(item.taskState)} />,
       },
+      {
+        key: 'retryCount',
+        name: 'Retry Count',
+        headerClassName: FontClassNames.medium,
+        minWidth: 100,
+        maxWidth: 100,
+        isResizable: true,
```
Comment (yqwang-ms) [inline src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx:636]: Can be just "Retries" like job? And reduce the witdth?

**Comment by yqwang-ms** [comment]
>         minWidth: 300,

Why not set all minWidth = 0, to save col space?

---
Refers to: src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx:650 in 7f204d6. [](commit_id = 7f204d6b1b0092318a5cb0a8522cc33ecc872e85, deletion_comment = False)

File: src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx (hunk 1 above)
Comment (debuggy) [reply src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx:346]: Prefer to keep exitCode, exitType, exitDiagonostic  as debug mode:
1. If they are default values, the detail list appearance will not that friendly for normal user
2. They actually belong to debug info, so maybe it is reasonable to keep them into debug mode.
3. Maybe startTime and completionTime could be in default mode if needed.

**Comment by debuggy** [comment]
> 
> 
> > ```
> >     minWidth: 300,
> > ```
> 
> Why not set all minWidth = 0, to save col space?
> 
> Refers to: src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx:650 in [7f204d6](https://github.com/microsoft/pai/commit/7f204d6b1b0092318a5cb0a8522cc33ecc872e85). [](commit_id = [7f204d6](https://github.com/microsoft/pai/commit/7f204d6b1b0092318a5cb0a8522cc33ecc872e85), deletion_comment = False)

```minWidth = 0``` works like ```auto``` and will take more space than expected

File: src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx (hunk 1 above)
Comment (yqwang-ms) [reply src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx:346]: User cares about each Task's exit info, they are even more important and precise than stderr and stdout (and any others).

If you run a cmd, won't you ignore the exit code?

Comment (yqwang-ms) [reply src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx:346]: You can show it very tight, such as:
```
ExitInfo
USER_STOP(-220)
```
And the ```USER_STOP(-220)``` is linked to Show Exit Diag

**Comment by yqwang-ms** [comment]
> > > ```
> > >     minWidth: 300,
> > > ```
> > 
> > 
> > Why not set all minWidth = 0, to save col space?
> > Refers to: src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx:650 in [7f204d6](https://github.com/microsoft/pai/commit/7f204d6b1b0092318a5cb0a8522cc33ecc872e85). [](commit_id = [7f204d6](https://github.com/microsoft/pai/commit/7f204d6b1b0092318a5cb0a8522cc33ecc872e85), deletion_comment = False)
> 
> `minWidth = 0` works like `auto` and will take more space than expected

How about minWidth = 1?

**Comment by debuggy** [comment]
image

**Comment by debuggy** [comment]
> How about minWidth = 1?

minWidth=1 will make the original column too tight that user could not see the title clearly.

**Comment by fanyangCS** [comment]
could you rename "Debug Info" to something else? For example, "Diagnosis Information"?

**Comment by yqwang-ms** [comment]
Move Debug Info  button to job level?

**Comment by debuggy** [comment]
updated：
image

File: src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx (hunk 1 above)
Comment (yqwang-ms) [reply src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx:346]: Teams call?

File: src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx [hunk 3]
```diff
@@ -321,29 +323,116 @@ export default class TaskRoleContainerList extends React.Component {
...
+            <div className={c(FontClassNames.mediumPlus)}>
+              {printDateTime(DateTime.fromISO(item.startTime))}
+            </div>
           );
         },
       },
       {
-        key: 'name',
-        name: 'Container ID',
+        key: 'completionTime',
```
Comment (yqwang-ms) [inline src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx:344]: Add debug info:
currentAttemptLaunchedTime
currentAttemptCompletedTime

Comment (debuggy) [reply src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx:359]: image

File: src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx [hunk 4]
```diff
@@ -321,29 +323,165 @@ export default class TaskRoleContainerList extends React.Component {
...
+          return (
+            <div className={c(FontClassNames.mediumPlus)}>
+              {printDateTime(DateTime.fromMillis(item.completedTime))}
+            </div>
+          );
+        },
+      },
+      {
+        key: 'currentAttemptLaunchedTime',
+        name: 'Current Attempt Launched Time',
```
Comment (yqwang-ms) [inline src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx:375]: Move currentAttemptLaunchedTime/currentAttemptCompletedTime between createdTime and completionTime?

File: src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx (hunk 3 above)
Comment (yqwang-ms) [reply src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx:359]: Just ```More Diagnostics```?

File: src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx (hunk 4 above)
Comment (debuggy) [reply src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx:360]: image

**Pull request merged**